### ISSUE TITLE
Compilation errors on LZMA (Linux)

### ISSUE THREAD

**Issue by RafaelPalomar** [Slicer/Slicer#6143]
I get compilation errors on LZMA on Gentoo Linux (gcc-11.2.0)

## Steps to reproduce

The building process stops with the following output:
```
[  1%] Building C object CMakeFiles/lzma.dir/common/block_buffer_encoder.c.o
In file included from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:32:9: error: unknown type name ‘CC_SHA256_CTX’
   32 | typedef CC_SHA256_CTX lzma_sha256_state;
      |         ^~~~~~~~~~~~~
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h: In function ‘lzma_sha256_init’:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:133:31: warning: passing argument 1 of ‘SHA256Init’ from incompatible pointer type [-Wincompatible-pointer-types]
  133 |         LZMA_SHA256FUNC(Init)(&check->state.sha256);
      |                               ^~~~~~~~~~~~~~~~~~~~
      |                               |
      |                               lzma_sha256_state * {aka int *}
In file included from /usr/include/sha256.h:32,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:22,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/usr/include/sha2.h:70:17: note: expected ‘SHA2_CTX *’ but argument is of type ‘lzma_sha256_state *’ {aka ‘int *’}
   70 | void SHA256Init(SHA2_CTX *);
      |                 ^~~~~~~~~~
In file included from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h: In function ‘lzma_sha256_update’:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:150:33: warning: passing argument 1 of ‘SHA256Update’ from incompatible pointer type [-Wincompatible-pointer-types]
  150 |         LZMA_SHA256FUNC(Update)(&check->state.sha256, buf, size);
      |                                 ^~~~~~~~~~~~~~~~~~~~
      |                                 |
      |                                 lzma_sha256_state * {aka int *}
In file included from /usr/include/sha256.h:32,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:22,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/usr/include/sha2.h:72:19: note: expected ‘SHA2_CTX *’ but argument is of type ‘lzma_sha256_state *’ {aka ‘int *’}
   72 | void SHA256Update(SHA2_CTX *, const uint8_t *, size_t);
      |                   ^~~~~~~~~~
In file included from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h: In function ‘lzma_sha256_finish’:
/home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:157:50: warning: passing argument 2 of ‘SHA256Final’ from incompatible pointer type [-Wincompatible-pointer-types]
  157 |         LZMA_SHA256FUNC(Final)(check->buffer.u8, &check->state.sha256);
      |                                                  ^~~~~~~~~~~~~~~~~~~~
      |                                                  |
      |                                                  lzma_sha256_state * {aka int *}
In file included from /usr/include/sha256.h:32,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/check/check.h:22,
                 from /home/rafael/src/Slicer/Slicer-master/build/LZMA/common/block_buffer_encoder.c:17:
/usr/include/sha2.h:74:50: note: expected ‘SHA2_CTX *’ but argument is of type ‘lzma_sha256_state *’ {aka ‘int *’}
   74 | void SHA256Final(uint8_t [SHA256_DIGEST_LENGTH], SHA2_CTX *);
      |                                                  ^~~~~~~~~~
gmake[5]: *** [CMakeFiles/lzma.dir/build.make:90: CMakeFiles/lzma.dir/common/block_buffer_encoder.c.o] Error 1
gmake[4]: *** [CMakeFiles/Makefile2:110: CMakeFiles/lzma.dir/all] Error 2
gmake[3]: *** [Makefile:136: all] Error 2
gmake[2]: *** [CMakeFiles/LZMA.dir/build.make:87: LZMA-prefix/src/LZMA-stamp/LZMA-build] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:717: CMakeFiles/LZMA.dir/all] Error 2
gmake: *** [Makefile:101: all] Error 2

Compilation exited abnormally with code 2 at Fri Jan 28 10:49:12
```

This happens in systems where `libmd` (e.g.,`libmd-dev` Ubuntu/Debian) is installed. This seems to be a bug in [slicer/lib_libzma ](https://github.com/Slicer/lib_lzma) and upstream [nextgis-borsch/lib_lzma](https://github.com/nextgis-borsch/lib_lzma). PRs to solve this issue have been proposed (slicer/lib_lzma#1 and nextgis-borsch/lib_lzma#2)

## Expected behavior

Build should complete successfully

## Workaround
One can configure Slicer with `-DSlicer_USE_SYSTEM_LZMA=ON` as a workaround.

## Environment
- Slicer version: Slicer-master
- Operating system:  Gentoo Linux (potentially other distros)

**Comment by RafaelPalomar** [comment]
@jcfr, what do you think about this?

**Comment by jcfr** [comment]
Following discussion:
* Remove `External_lzma.cmake` and add `External_xz.cmake` based of https://github.com/xz-mirror/xz

**Comment by RafaelPalomar** [comment]
This can be closed based on slicer/slicer#6236. In summary:

  - `LZMA` is part of the `xz` package.
  - The version used in Slicer was a CMakefied `LZMA` extracted and separated from `xz`, but it wasn't well maintained.
  - slicer/slicer#6236 replaces the old `LZMA` by `xz` and updates to the latest version (v5.2.2 earlier, now v5.2.5).
  - `xz` v5.2.5 solves the compilation problem in this issue
